services/main/Controllers/Waitlist.cs

```diff
@@ -0,0 +1,34 @@
+
+using main.DTOs;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+namespace main.Controllers;
+
+[ApiController]
+[Route("api/v1/waitlists")]
+public class WaitlistController : ControllerBase
+{
+    private readonly ILogger<WaitlistController> logger;
+    private readonly WaitlistService waitlistService;
+
+    public WaitlistController(ILogger<WaitlistController> logger, WaitlistService waitlistService)
+    {
+        this.logger = logger;
+        this.waitlistService = waitlistService;
+    }
+
+    [HttpPost("create")]
+    public async Task<ActionResult> AddWaitlistUser([FromBody] WaitlistEntry waitlistEntry)
+    {
+        try
+        {
+            await this.waitlistService.SaveWaitlistDetails(waitlistEntry);
+            return this.Ok();
+        } 
+        catch (Exception e)
+        {
+            this.logger.LogError($"Failed to save waitlist entry into database. Exception: {e}");
+            return new StatusCodeResult(500);
+        }
+    }
+}
```

Comment:
```suggestion
 catch (HttpRequestException e)
        {
            // error response
        }
        catch (Exception e)
        {
            this.logger.LogError($"Failed to save waitlist entry into database. Exception: {e}");
            return new StatusCodeResult(500);
        }
    }
}
```
 Also,
- we use `HttpRequestException` to catch any "user defined" errors that we throw in services. This helps us set custom status codes like 400, 422, etc. Frontend heavily makes use of these status codes as well. So once we send the wrong status code, it impacts them as well
- Reason for separating the `Exception` is because we want to catch errors that shouldn't happen; Any error that was out of our control in code. We'll be setting up an error logger like Sentry to send notifications when these errors come up in the future